### PR TITLE
Fix SIM sensor crash

### PR DESCRIPTION
**Pull request by dshokouhi**
## Summary

Fixes: #2386 by checking that display name is not null and also catching any other potential failures

## Link to pull request in Documentation repository

Documentation: home-assistant/companion.home-assistant#

### CODE REVIEW

File: app/src/main/java/io/homeassistant/companion/android/sensors/PhoneStateSensorManager.kt [hunk 1]
```diff
@@ -121,16 +122,22 @@ class PhoneStateSensorManager : SensorManager {
...
+                        attrs["carrier name"] = info.carrierName
+                        attrs["iso country code"] = info.countryIso
+                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+                            attrs["carrier id"] = info.carrierId
+                            attrs["mcc"] = info.mccString.toString()
+                            attrs["mnc"] = info.mncString.toString()
+                            attrs["is opportunistic"] = info.isOpportunistic
+                            if (info.dataRoaming == SubscriptionManager.DATA_ROAMING_ENABLE) attrs["data roaming"] =
+                                "enable"
+                            else attrs["data roaming"] = "disable"
```
Comment (JBassett) [inline app/src/main/java/io/homeassistant/companion/android/sensors/PhoneStateSensorManager.kt:137]:
```suggestion
                             attrs["data roaming"] = if (info.dataRoaming == SubscriptionManager.DATA_ROAMING_ENABLE) "enable" else "disable"
```

File: app/src/main/java/io/homeassistant/companion/android/sensors/PhoneStateSensorManager.kt [hunk 2]
```diff
@@ -121,16 +122,22 @@ class PhoneStateSensorManager : SensorManager {
...
-                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
-                        attrs["carrier id"] = info.carrierId
-                        attrs["mcc"] = info.mccString.toString()
-                        attrs["mnc"] = info.mncString.toString()
-                        attrs["is opportunistic"] = info.isOpportunistic
-                        if (info.dataRoaming == SubscriptionManager.DATA_ROAMING_ENABLE) attrs["data roaming"] = "enable"
-                        else attrs["data roaming"] = "disable"
+                    try {
+                        if (info.displayName != null)
+                            displayName = info.displayName.toString()
```
Comment (JBassett) [inline app/src/main/java/io/homeassistant/companion/android/sensors/PhoneStateSensorManager.kt:127]:
```suggestion
                        displayName = info.displayName?.toString() ?: displayName
```

**Review by JBassett — CHANGES_REQUESTED** [review]

**Comment by jpelgrom** [comment]
FYI, I've opened another PR #2391 to prevent crashing the UI as well. Still a good idea to catch and handle errors in the manager itself though :)

**Comment by dshokouhi** [comment]
> FYI, I've opened another PR #2391 to prevent crashing the UI as well. Still a good idea to catch and handle errors in the manager itself though :)

I agree we need it in both places, good idea!

**Review by JBassett — APPROVED** [review]

**Pull request merged**